### PR TITLE
refactor: Use timezone-aware datetime objects

### PR DESCRIPTION
**Pull request by edgarrmondragon**

### CODE REVIEW

**Comment by WillDaSilva** [comment]
@edgarrmondragon Just curious, why close this?

**Comment by edgarrmondragon** [comment]
> @edgarrmondragon Just curious, why close this?

Noticed the attempt was too ambitious and broke some tests 😅. Looking back, it's only 8 tests so probably worth re-opening and trying to revert the offending changes.

File: src/meltano/core/sqlalchemy.py [hunk 1]
```diff
@@ -94,6 +95,42 @@ def process_result_value(  # noqa: D102
         return value
 
 
+class DateTimeUTC(TypeDecorator):
```
Comment (WillDaSilva) [inline src/meltano/core/sqlalchemy.py:98]: TIL

File: tests/meltano/core/test_project_files.py [hunk 2]
```diff
@@ -182,23 +182,23 @@ def test_load(self, project_files):
                     "extractor": "tap-meltano-yml",
                     "loader": "target-meltano-yml",
                     "transform": "skip",
-                    "start_date": datetime.datetime(2020, 8, 5),  # noqa: WPS432
+                    "start_date": datetime.datetime(2020, 8, 5),  # noqa: DTZ001
```
Comment (WillDaSilva) [inline tests/meltano/core/test_project_files.py:185]: Why not add the tzinfo for these?

**Review by WillDaSilva — APPROVED** [review]

File: tests/meltano/core/test_project_files.py (hunk 2 above)
Comment (edgarrmondragon) [reply tests/meltano/core/test_project_files.py]: 5213ebd2062b912066854b8a3fc578e73755f008

Comment (edgarrmondragon) [reply tests/meltano/core/test_project_files.py]: 5213ebd2062b912066854b8a3fc578e73755f008

**Pull request merged**